blog/content/post/2020-12-08-IOMMU.md

```diff
@@ -0,0 +1,87 @@
+---
+title: What is IOMMU and how it can be used?
+abstract: 'Introductory blog post on how IOMMU works.
+          In this article you can read what IOMMU is
+          and find out if its use may be beneficial for you.'
+cover: /covers/IOMMU.png
+author: marek.kasiewicz
+layout: post
+published: true
+date: 2020-12-08
+archives: "2020"
+
+tags:
+  - Firmware
+  - IOMMU
+  - Security
+categories:
+  - Firmware
+  - Security
+
+---
+
+## Introduction
+
+The I/O memory management unit (IOMMU) is a type of memory management unit (MMU)
+that connects a Direct Memory Access (DMA) capable expansion bus to the main memory.
+It extends the system architecture by adding support for the virtualization of
+memory addresses used by peripheral devices. Additionally, it provides memory
+isolation and protection by enabling system software to control which areas of
+physical memory an I/O device may access. It also helps filter and remap interrupts
+from peripheral devices. 
```

Comment:
Let's have a look at IOMMU advantages, disadvantages, and purpose in both, hardware and software.